test/src/interfacekernels/PenaltyInterfaceDiffusionDot.C

```diff
@@ -0,0 +1,75 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PenaltyInterfaceDiffusionDot.h"
+
+registerMooseObject("MooseTestApp", PenaltyInterfaceDiffusionDot);
+
+template <>
+InputParameters
+validParams<PenaltyInterfaceDiffusionDot>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addRequiredParam<Real>(
+      "penalty", "The penalty that penalizes jump between master and neighbor variables.");
+  return params;
+}
+
+PenaltyInterfaceDiffusionDot::PenaltyInterfaceDiffusionDot(const InputParameters & parameters)
+  : InterfaceKernel(parameters), _penalty(getParam<Real>("penalty"))
+{
+}
+
+Real
+PenaltyInterfaceDiffusionDot::computeQpResidual(Moose::DGResidualType type)
+{
+  Real r = 0;
+
+  switch (type)
+  {
+    case Moose::Element:
+      r = _test[_i][_qp] * _penalty * (_u_dot[_qp] - _neighbor_value_dot[_qp]);
+      break;
+
+    case Moose::Neighbor:
+      r = _test_neighbor[_i][_qp] * -_penalty * (_u_dot[_qp] - _neighbor_value_dot[_qp]);
+      break;
+  }
+
+  return r;
+}
+
+Real
+PenaltyInterfaceDiffusionDot::computeQpJacobian(Moose::DGJacobianType type)
```

Comment:
I'm can't figure out what's wrong with the Jacobian.

Reply:
Did you try PJFNK solve_type?

Reply:
@YaqiWang, yes I tried and it works. However, I always try to use newton to make sure I have the right jacobian. The thing is newton works with FDP but not with the supplied, ideally exact, jacobian. So it should be messed-up somehow

Reply:
Ok, I figured it out. I t was my mistake in defying _u_dot(_var.uDot()),